rpc/rpcserver/server.go

```diff
@@ -1161,6 +1169,133 @@ func StartWalletLoaderService(server *grpc.Server, loader *loader.Loader,
 	pb.RegisterWalletLoaderServiceServer(server, service)
 }
 
+// StartTicketBuyerService creates an implementation of the TicketBuyerService
+// and registers it with the gRPC server.
+func StartTicketBuyerService(server *grpc.Server, loader *loader.Loader,
+	tbCfg *ticketbuyer.Config) {
+
+	service := &ticketbuyerLoaderServer{loader: loader, ticketbuyerCfg: tbCfg}
+	pb.RegisterTicketBuyerServiceServer(server, service)
+}
+
+// StartTicketPurchase enables ticket purchases in the ticket buyer.
+func (t *ticketbuyerLoaderServer) StartTicketPurchase(ctx context.Context, req *pb.StartTicketPurchaseRequest) (
+	*pb.StartTicketPurchaseResponse, error) {
+	defer zero.Bytes(req.Passphrase)
+
+	wallet, ok := t.loader.LoadedWallet()
+	if !ok {
+		return nil, grpc.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+	if req.Passphrase != nil {
+		err := wallet.Unlock(req.Passphrase, nil)
+		if err != nil {
+			return nil, translateError(err)
+		}
+	}
+
+	accountName, err := wallet.AccountName(req.Account)
+	if err != nil {
+		return nil, translateError(err)
+	}
+
+	balanceToMaintain := dcrutil.Amount(req.BalanceToMaintain)
+	if balanceToMaintain < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative balance to maintain given")
+	}
+
+	maxFeePerKB := dcrutil.Amount(req.MaxFeePerKb)
+
+	if maxFeePerKB < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative max fee per KB given")
+	}
+
+	if req.MaxPriceAbsolute < 0 && req.MaxPriceRelative < 0 {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Negative max ticket price given")
+	}
+	params := wallet.ChainParams()
+
+	var votingAddress string
+	if req.VotingAddress != "" {
+		votingAddr, err := decodeAddress(req.VotingAddress, params)
+		if err != nil {
+			return nil, err
+		}
+		votingAddress = votingAddr.String()
+	}
+
+	var poolAddress string
+	if req.PoolAddress != "" {
+		poolAddr, err := decodeAddress(req.PoolAddress, params)
+		if err != nil {
+			return nil, err
+		}
+		poolAddress = poolAddr.String()
+	}
+
+	poolFees := req.PoolFees
+	if poolFees > 0 {
+		err = txrules.IsValidPoolFeeRate(poolFees)
+		if err != nil {
+			return nil, grpc.Errorf(codes.InvalidArgument,
+				"Pool fees amount invalid: %v", err)
+		}
+	}
+
+	if poolFees > 0 && poolAddress == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument,
+			"Pool fees set but no pool address given")
+	}
+
+	if poolFees <= 0 && poolAddress != "" {
```

Comment:
looks like there's a missing check for negative pool fees when the address is the empty string